mautic/mautic
# Workaround for lost campaign segments after clone campaign

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Yes
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5682
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This issue is reported https://github.com/mautic/mautic/issues/5682 
After clone campaign, if you launch builder and edit segments source, the campaign during save entity, lost segment source. Browser return log: **Cannot establish connection - source does not exist**
This workaround don't allow  launch builder until you save cloned campaign.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:

1. Clone a campaign that starts with a segment
2. Go on the builder
3. Click on the segment, do not change it (keep the same segment) but just click on update
4. Close the builder, save and close the campaign.
5. Go back on the campaign and go back on the builder,
6. You will see that there are no segments anymore

#### Steps to test this PR:
1.  Apply PR and clone campaign
2. Try Launch Campaign Builder
3.  See  alert and no campaign builder launched

![image](https://user-images.githubusercontent.com/462477/36489515-83bcab1a-1726-11e8-97b2-07defdf6e37c.png)
